### PR TITLE
Fix relative links in README descriptions

### PR DESCRIPTION
**Pull request by kathy-t**
**Description**
This PR fixes the relative links displayed in an entry's README description by specifying a base URL for relative links. 

ngx-markdown allows options like `baseUrl` (prefix url for relative links) to be specified as [MarkedOptions](https://github.com/jfcere/ngx-markdown#markedoptions). However, MarkedOptions is meant to be defined globally, which isn't ideal for us because relative links only really show up in README descriptions and not in other places where we use markdown. Setting it globally also prevents us from changing the base URL dynamically based on entry repo.

Found a work-around by passing in MarkedOptions in the MarkDownService's `compile` method, which we already use. My PR only specifies base URLs for tool and workflow descriptions.

I found that GitLab and BitBucket also support relative links in their READMEs, so this PR handles those services as well.

**Issue**
[#4453](https://github.com/dockstore/dockstore/issues/4453)

Please make sure that you've checked the following before submitting your pull request. Thanks!

- [x] Check that your code compiles by running `npm run build`
- [x] If this is the first time you're submitting a PR or even if you just need a refresher, consider reviewing our [style guide](https://github.com/dockstore/dockstore/wiki/Dockstore-Frontend-Opinionated-Style-Guide#pr-checklist)
- [x] Do not bypass Angular sanitization (bypassSecurityTrustHtml, etc.), or justify why you need to do so
- [x] If displaying markdown, use the `markdown-wrapper` component, which does extra sanitization
- [x] Do not use cookies, although this may change in the future
- [x] Run `npm audit` and ensure you are not introducing new vulnerabilities
- [x] Do due diligence on new 3rd party libraries, checking for CVEs
- [x] Don't allow user-uploaded images to be served from the Dockstore domain

### CODE REVIEW

File: src/app/shared/entry/base-url.pipe.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+export class BaseUrlPipe implements PipeTransform {
+  /**
+   * Constructs a base url for relative links in markdown README files.
+   * @param providerUrl
+   * @param versionName
+   * @returns {string} base url to be used by ngx-markdown
+   */
+  transform(providerUrl: string, versionName: string): string {
+    if (!providerUrl || !versionName) {
+      console.error('providerUrl or versionName is not truthy');
```
Comment (coverbeck) [inline src/app/shared/entry/base-url.pipe.ts:15]: I'm not a fan of logging to the user's browser console, but others disagree. :)

**Review by coverbeck — APPROVED** [review]
Added @david4096 as a reviewer for any security implications. Trying to think if there's some way a user could create a malicious README.md. The existing code already prevents scripts from running. I guess there could be evil link, but somebody could do that already with an absolute url.

I can't think of anything.

**Review by garyluu — APPROVED** [review]

**Review by NatalieEO — APPROVED** [review]

**Review by denis-yuen — COMMENTED** [review]
No objections on my end, but curious about the security discussion

**Review by Richard-Hansen — APPROVED** [review]

**Review by david4096 — APPROVED** [review]
I can't think of anything off hand that someone could take advantage of here that the absolute URL wouldn't also affect. Looks good to me

**Review by denis-yuen — APPROVED** [review]

**Pull request merged**